src/devices/machine/smartmed.h

```diff
@@ -156,7 +156,8 @@ class nand_device : public device_t
 						// 0 means no card loaded
 	int m_log2_pages_per_block; // log2 of number of pages per erase block (usually 4 or 5)
 
-	uint8_t* m_data_ptr;  // FEEPROM data area
+	uint8_t * m_data_ptr;  // FEEPROM data area
+	std::unique_ptr<uint8_t[]> m_data_ptr_alloc;
```

Comment:
I was going for the minimal necessary changes here. Would you prefer I rename things and expand the diff?

Reply:
In case it wasn’t obvious from the first comment, I prefer the code to be in a better state afterwards, not “shut up warnings with as few changed lines as possible”.  If you make the member names confusing, it requires another person to go through and clean up after you/

Reply:
Ok